### PR TITLE
test: add simple typing tests

### LINKED ISSUE #199: Revisit the type changes
We should revisit the type changes reverted in #198   Mostly, we should have some typescript tests to make sure those new types are not breaking anything. 

Related to #195

### PR DESCRIPTION
**Pull request by mcous**
## Overview

This PR adds simple typing tests. Since Vitest's type-testing doesn't support Svelte (only vanilla TS and Vue), I implemented tests using [expect-type](https://github.com/mmkal/expect-type) - the library Vitest uses under the hood - and [svelte-check](https://github.com/sveltejs/language-tools/tree/master/packages/svelte-check).

Closes #199, related to #194, #195

## Change log

- Add simple typing test for render function
- Lint and typecheck in CI for latest Svelte + latest Node.js
    - Skip linting/typechecking otherwise

### CODE REVIEW

**Comment by mcous** [comment]
I was really hoping I could come up with a clever fix for this, but no such luck. I'll disable in the matrix as suggested.

I wonder what this looks like in an actual Scelte 3 + TS project? Might have to spin one up and check it out; probably worth noting in the docs while I'm in there

**Comment by mcous** [comment]
Spun up a little Svelte 3 + TS project, and I'm not 100% sure how, but the types definitely work correctly. Probably some magic from `svelte-check`.

Interestingly enough, while testing I discovered that the latest version of `@sveltejs/vite-plugin-svelte` has dropped support for Svelte 3. This will likely force our hand at some point, as much as I'd prefer not to drop support here. But for now, I can be comfortable with the fact that we only check types using Svelte 4 in CI

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -34,8 +34,14 @@ jobs:
...
 
-      - name: ▶️ Run validate script
-        run: npm run validate
+      - name: ▶️ Test
+        run: npm test
+
+      - name: ▶️ Lint and type-check
+        if: ${{ matrix.node == '20' && matrix.svelte == '4' }}
+        run: |
+          npm run lint
```
Comment (yanick) [inline .github/workflows/release.yml:43]: As per the other discussion, we can't assume all files will be formatted to prettier's and eslint's current expectations.

```suggestion
      - name: ▶️ type-check
        if: ${{ matrix.node == '20' && matrix.svelte == '4' }}
        run: |
```

File: package.json [hunk 2]
```diff
@@ -86,12 +87,18 @@
     "eslint-plugin-simple-import-sort": "10.0.0",
     "eslint-plugin-svelte": "2.35.1",
     "eslint-plugin-vitest-globals": "1.4.0",
+<<<<<<< HEAD
```
Comment (mcous) [inline package.json:90]: Well this is embarrassing

**Review by mcous — COMMENTED** [review]
Nothing to see here...

**Pull request merged**